### ISSUE TITLE
[4.0.0-rc05] Repeat mode does not work on iOS

### ISSUE THREAD

**Issue by TobiDevloft** [doublesymmetry/react-native-track-player#2070]
**Describe the Bug**
Repeat mode causes very strange and inconsistent behavior on iOS.

On the emulator, everything works as expected: A track finishes and then restarts as expected.

On a real device however, this does not happen. Once the track is finished, instead of repeating the track the player goes into a paused state. When that happens and I switch the track, the state changes but sometimes the progress just doesn't get updated (though this part is not reproduceible consistently). 

I also experience progress updates getting fired with position set to below zero (e.g. -0.51), but I figure that is a different issue.

I opened https://github.com/doublesymmetry/react-native-track-player/issues/1995 back in April, but that was for `4.0.0-rc05`. There I incorrectly stated, that the issue does not appear anymore in `rc05`. It does appear, but I was working with an emulator where is works as expected.

**Steps To Reproduce**
Set the player to `RepeatMode.Track `on a real iOS device.

**Code To Reproduce**
`await TrackPlayer.setRepeatMode(RepeatMode.Track)`

**Replicable on Example App?**
Yes.

**Environment Info:**
```
System:
    OS: macOS 13.2.1
    CPU: (12) arm64 Apple M2 Pro
    Memory: 255.80 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.10.0 - ~/.nvm/versions/node/v16.10.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.10.0/bin/yarn
    npm: 7.24.0 - ~/.nvm/versions/node/v16.10.0/bin/npm
    Watchman: 2023.05.22.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.12.0 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    Android SDK:
      Android NDK: 25.2.9519653
  IDEs:
    Android Studio: Not Found
    Xcode: 14.3/14E222b - /usr/bin/xcodebuild
  Languages:
    Java: 16.0.2 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0
    react-native: 0.71.7 => 0.71.7
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Paste the exact react-native-track-player version you are using

4.0.0-rc5

Real device? Or simulator?

Real device

What OS are you running?

iOS 16.4.1

**How I can Help**
What can you do to help resolve this?
If you point me in the right direction, I'll try my best but I only have very limited swift experience.

**Comment by TobiDevloft** [comment]
Hey @dcvz, we've recently did some tests on version 4.0.0-rc06 and, although the issue doesn't occur consistently, we still encounter it from time to time. Unfortunately, we haven't identified a reliable way to reproduce it. Approximately every 5-10 attempts, the player pauses, and the playback state transitions from loading to paused to ready.

Additionally, we've observed a separate issue on iOS, where progress updates occasionally show negative values (e.g., -0.286526) after repeating a track. Upon restarting, the progress update is called multiple times in quick succession, and the sequence looks like this (example): 0, 0, 0, 0, -0.22, -0.22, 1. I'm not sure if this is directly related to the repeat mode issue, but I thought I'd share it as it may provide some useful insights.

**Comment by kirillzyusko** [comment]
@TobiDevloft @jr00n I added a fix/workaround in https://github.com/doublesymmetry/SwiftAudioEx/pull/62 - if you have time and desire you can test it 👀

**Comment by TobiDevloft** [comment]
@kirillzyusko @dcvz This seems to have fixed it! Thank you so much for fixing this, you've saved me a bunch of headaches haha

**Comment by KaiValar** [comment]
This same bug is present in 4.1.1 if you use `RepeatMode.Queue`
